addons/website/static/src/snippets/s_facebook_page/options.js

```diff
@@ -156,22 +187,29 @@ options.registry.facebookPage = options.Class.extend({
         // The regex is kept as a huge one-liner for performance as it is
         // compiled once on script load. The only way to split it on several
         // lines is with the RegExp constructor, which is compiled on runtime.
-        const match = this.fbData.href.match(/^(https?:\/\/)?((www\.)?(fb|facebook)|(m\.)?facebook)\.com\/(((profile\.php\?id=|people\/[^/?#]+\/|(p\/)?[^/?#]+-)(?<id>[0-9]{15,16}))|(?<nameid>[\w.]+))($|[/?# ])/);
+        const match = this.fbData.href.trim().match(/^(https?:\/\/)?((www\.)?(fb|facebook)|(m\.)?facebook)\.com\/(((profile\.php\?id=|people\/[^/?#]+\/|(p\/)?[^/?#]+-)(?<id>[0-9]{15,16}))|(?<nameid>[\w.]+))($|[/?# ])/);
```

Comment:
Note that for URLs like `https://www.facebook.com/my.super.page`, the problem is not our checker, it's the URL from FB Graph we use to check if it's valid: those URLs return an error. So unless we find a better FB graph checker, that's a hard limitation.

Reply:
Here are the tests I've been running and determining why I extracted the logic of parsing in a private function
https://excalidraw.com/#json=O2zFp0uwBeMbk--08WsOH,lKjpQWLrNHrNrJoD3TkIzA

The real interest is to properly recover the identifier after the regex has validated the URL because the regex does two things :
 - identify a valid URL, the regex does it well
 - parse the id/nameid, the regex doesn't do it that well in certain cases
 
 It felt to me the good way of doing that was to let each part do its own job, the regex validates the URL and the function parse the identifier. Is it clearer? Maybe I may have the wrong approach...